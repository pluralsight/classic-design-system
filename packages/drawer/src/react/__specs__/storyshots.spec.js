import initStoryshots, { snapshotWithOptions } from '@storybook/addon-storyshots'

function createNodeMock(element) {
  if (element.type === 'div') {
    return { style: {} }
  }
  return null
}

initStoryshots({
  test: snapshotWithOptions({
    createNodeMock
  })
})
