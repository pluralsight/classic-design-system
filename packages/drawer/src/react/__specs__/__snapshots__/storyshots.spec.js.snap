// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots drawer controlled 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <button
        data-css-bzft25=""
        disabled={false}
        onClick={[Function]}
      >
        <span
          className="css-1en8fmq"
        >
          Toggle drawer
        </span>
      </button>
      <div>
        <div
          className="css-a8p3d5"
          onClick={[Function]}
        >
          <div
            className="css-1lzobj4"
          >
            <div
              className="css-1g6ond4"
            >
              <div>
                Drawer Base Here
              </div>
            </div>
          </div>
          <div
            className="css-qs2rg7"
          >
            <button
              className="css-wepra9"
              onClick={[Function]}
            >
              <div
                className="css-xnveog"
              >
                <div
                  className="css-otejxm"
                >
                  <svg
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          className="css-ule0ur"
        >
          <div
            className="css-8nylan"
          >
            Drawer Content Here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer default 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <div
        className="css-58omaf"
        onClick={[Function]}
      >
        <div
          className="css-1lzobj4"
        >
          <div
            className="css-1g6ond4"
          >
            <div>
              Drawer Base Here
            </div>
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            className="css-wepra9"
            onClick={[Function]}
          >
            <div
              className="css-1byjzqg"
            >
              <div
                className="css-otejxm"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-1i2htk9"
      >
        <div
          className="css-8nylan"
        >
          Drawer Content Here
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer row component with actions 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <div
        className="css-58omaf"
        onClick={[Function]}
      >
        <div
          className="css-1lzobj4"
        >
          <div
            className="css-1g6ond4"
          >
            <div
              className="css-7w4714"
            >
              <div
                className="css-js3ti3"
              >
                <span
                  className="css-15yq56v"
                >
                  <a
                    href="https://duckduckgo.com?q=image"
                  >
                    <img
                      src="https://cataas.com/cat"
                    />
                  </a>
                </span>
                <div
                  className="css-4nxbcy"
                >
                  <span
                    className="css-1fxa3fl"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    <a
                      href="https://duckduckgo.com?q=overlay"
                    >
                      Overlay
                    </a>
                  </span>
                </div>
              </div>
              <div
                className="css-kxulf3"
              >
                <div
                  className="css-1e23mlv"
                >
                  <span
                    className="css-1ublh7e"
                  >
                    <a
                      href="https://duckduckgo.com?q=title"
                    >
                      I'm a Row with Actions
                    </a>
                  </span>
                </div>
                <div
                  className="css-gmo5z6"
                >
                  <span
                    className="css-1tz10dh"
                  >
                    <span
                      className="css-1ublh7e"
                    >
                      <a
                        href="https://duckduckgo.com?q=cats"
                      >
                        Kitten McCatbuns
                      </a>
                    </span>
                  </span>
                  <span
                    aria-hidden={true}
                    className="css-kykfz1"
                  >
                    Â·
                  </span>
                  <span
                    className="css-1tz10dh"
                  >
                    23 hours of cuteness
                  </span>
                </div>
              </div>
              <div
                className="css-1keoilf"
              >
                <button
                  className="css-1eg2eyk"
                  onClick={[Function]}
                >
                  <div
                    className="css-otejxm"
                  >
                    <svg
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M6 14.5c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2zm12 0c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2zm-6 0c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2z"
                      />
                    </svg>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            className="css-wepra9"
            onClick={[Function]}
          >
            <div
              className="css-1byjzqg"
            >
              <div
                className="css-otejxm"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-1i2htk9"
      >
        <div
          className="css-8nylan"
        >
          Drawer Content Here
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer using onToggle prop 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <div>
        <div
          className="css-58omaf"
          onClick={[Function]}
        >
          <div
            className="css-1lzobj4"
          >
            <div
              className="css-1g6ond4"
            >
              <div>
                Drawer is 
                closed
              </div>
            </div>
          </div>
          <div
            className="css-qs2rg7"
          >
            <button
              className="css-wepra9"
              onClick={[Function]}
            >
              <div
                className="css-1byjzqg"
              >
                <div
                  className="css-otejxm"
                >
                  <svg
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          className="css-1i2htk9"
        >
          <div
            className="css-8nylan"
          >
            Drawer Content Here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer using startOpen prop 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <div
        className="css-a8p3d5"
        onClick={[Function]}
      >
        <div
          className="css-1lzobj4"
        >
          <div
            className="css-1g6ond4"
          >
            <div>
              Base here
            </div>
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            className="css-wepra9"
            onClick={[Function]}
          >
            <div
              className="css-xnveog"
            >
              <div
                className="css-otejxm"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-ule0ur"
      >
        <div
          className="css-8nylan"
        >
          <div>
            Panel here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer with row component 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <div
        className="css-58omaf"
        onClick={[Function]}
      >
        <div
          className="css-1lzobj4"
        >
          <div
            className="css-1g6ond4"
          >
            <div
              className="css-7w4714"
            >
              <div
                className="css-js3ti3"
              >
                <div
                  className="css-y8jh93"
                />
              </div>
              <div
                className="css-kxulf3"
              >
                <div
                  className="css-1e23mlv"
                >
                  Look at me! I'm a &lt;Row /&gt;!
                </div>
                <div
                  className="css-gmo5z6"
                >
                  <span
                    className="css-1tz10dh"
                  >
                    Kitten McCatbuns
                  </span>
                  <span
                    aria-hidden={true}
                    className="css-kykfz1"
                  >
                    Â·
                  </span>
                  <span
                    className="css-1tz10dh"
                  >
                    23 hours of cuteness
                  </span>
                </div>
              </div>
              <div
                className="css-1bg3chk"
              >
                <button
                  className="css-1eg2eyk"
                >
                  <div
                    className="css-otejxm"
                  >
                    <svg
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M6 14.5c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2zm12 0c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2zm-6 0c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2z"
                      />
                    </svg>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            className="css-wepra9"
            onClick={[Function]}
          >
            <div
              className="css-1byjzqg"
            >
              <div
                className="css-otejxm"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-1i2htk9"
      >
        <div
          className="css-8nylan"
        >
          Drawer Content Here
        </div>
      </div>
    </div>
  </div>
</div>
`;
